core/arch/riscv64/mangle.c

```diff
@@ -385,13 +394,406 @@ mangle_rel_addr(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
     return NULL;
 }
 
-instr_t *
-mangle_reads_thread_register(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
+static reg_id_t
+pick_scratch_reg(dcontext_t *dcontext, instr_t *instr, reg_id_t do_not_pick,
+                 ushort *scratch_slot OUT)
+{
+    reg_id_t reg = REG_NULL;
+    ushort slot = 0;
+
+    for (reg = SCRATCH_REG0, slot = TLS_REG0_SLOT; reg <= SCRATCH_REG_LAST;
+         reg++, slot += sizeof(reg_t)) {
+        if (!instr_uses_reg(instr, reg) && reg != do_not_pick)
+            break;
+    }
+
+    if (scratch_slot != NULL)
+        *scratch_slot = slot;
+
+    ASSERT(reg != REG_NULL);
+    return reg;
+}
+
+static void
+mangle_stolen_reg_and_tp_reg(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
                              instr_t *next_instr)
 {
-    /* FIXME i#3544: Not implemented */
-    ASSERT_NOT_IMPLEMENTED(false);
-    return false;
+    ushort slot;
```

Comment:
Add ASSERT_NOT_TESTED(), which can be removed once we have an actual test.  That makes it clear this has not actually been testing.  (Or if you are able to run an end to end test: please include that test in this PR.)